### PR TITLE
DATAES-593 - Add support for collapse field on SearchRequest.

### PR DESCRIPTION
**Pull request by chocho01**
Support for use collapse field function in the Query Apis

- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATAES).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@chocho01 Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-elasticsearch&pullRequestId=290)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-data-elasticsearch&pullRequestId=290) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@chocho01 Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-elasticsearch&pullRequestId=290)!

File: src/test/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplateTests.java [hunk 1]
```diff
@@ -2777,6 +2778,36 @@ public void shouldSortResultsGivenSortCriteriaFromPageableWithScanAndScroll() {
 		assertThat(sampleEntities.get(2).getMessage()).isEqualTo(sampleEntity1.getMessage());
 	}
 
+	@Test // DATAES-393
```
Comment (sothawo) [inline src/test/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplateTests.java]:
```suggestion
	@Test // DATAES-593
```

File: src/main/java/org/springframework/data/elasticsearch/core/query/NativeSearchQuery.java [hunk 2]
```diff
@@ -111,6 +113,15 @@ public void addScriptField(ScriptField... scriptField) {
         scriptFields.addAll(Arrays.asList(scriptField));
     }
 
+	@Override
+	public CollapseBuilder getCollapse() {
```
Comment (sothawo) [inline src/main/java/org/springframework/data/elasticsearch/core/query/NativeSearchQuery.java]:
```suggestion
	public CollapseBuilder getCollapseBuilder() {
```
the field is `collapseBuilder` the type is a Builder, so I think this method name should reflect this

File: src/main/java/org/springframework/data/elasticsearch/core/query/NativeSearchQuery.java [hunk 3]
```diff
@@ -111,6 +113,15 @@ public void addScriptField(ScriptField... scriptField) {
...
+		return collapseBuilder;
+	}
+
+	public void setCollapse(CollapseBuilder collapse) {
```
Comment (sothawo) [inline src/main/java/org/springframework/data/elasticsearch/core/query/NativeSearchQuery.java]:
```suggestion
	public void setCollapseBuilder(CollapseBuilder collapseBuilder) {
```

File: src/main/java/org/springframework/data/elasticsearch/core/query/NativeSearchQuery.java [hunk 4]
```diff
@@ -111,6 +113,15 @@ public void addScriptField(ScriptField... scriptField) {
...
+		this.collapseBuilder = collapse;
```
Comment (sothawo) [inline src/main/java/org/springframework/data/elasticsearch/core/query/NativeSearchQuery.java]:
```suggestion
		this.collapseBuilder = collapseBuilder;
```

**Comment by chocho01** [comment]
@sothawo Thanks for your feedback. Your suggestions have been integrated in previous commit and I push a new commit to add the functionality to the `ReactiveElasticsearchTemplate` class

**Comment by sothawo** [comment]
Your PR has been merged, thank you for your contribution!

**Pull request merged**